<launch>
  <arg name="port"                      default="8765" />
  <arg name="address"                   default="0.0.0.0" />
  <arg name="tls"                       default="false" />
  <arg name="certfile"                  default="" />
  <arg name="keyfile"                   default="" />
  <arg name="topic_whitelist"           default="'.*'" />
  <arg name="max_qos_depth"             default="10" />
  <arg name="num_threads"               default="0" />
  <arg name="use_sim_time"              default="false" />

  <node pkg="foxglove_bridge" exec="foxglove_bridge">
    <param name="port"                  value="$(var port)" />
    <param name="address"               value="$(var address)" />
    <param name="tls"                   value="$(var tls)" />
    <param name="certfile"              value="$(var certfile)" />
    <param name="keyfile"               value="$(var keyfile)" />
    <param name="topic_whitelist"       value="$(var topic_whitelist)"      value-sep="," />
    <param name="max_qos_depth"         value="$(var max_qos_depth)" />
    <param name="num_threads"           value="$(var num_threads)" />
    <param name="use_sim_time"          value="$(var use_sim_time)" />
  </node>
</launch>
